bulk_index.py

```diff
@@ -0,0 +1,106 @@
+"""Use this to populate a search index in bulk or prefetch metadata."""
+
+import os
+import click
+from search.factory import create_ui_web_app
+from search.agent import MetadataRecordProcessor, DocumentFailed, \
+    IndexingFailed
+
+app = create_ui_web_app()
+app.app_context().push()
+
+
+def chunks(l, n):
+    """Yield successive n-sized chunks from l."""
+    for i in range(0, len(l), n):
+        yield l[i:i + n]
```

Comment:
You might have a look at https://docs.python.org/3/library/itertools.html#itertools.tee -- I am constantly (re)discovering goodies in itertools

Reply:
Thanks, lots of goodies indeed! `itertools.tee` should work fine here.